### DISCUSSION TITLE
A palette with geometric shapes

### DISCUSSION THREAD

**Discussion by albesco** [eclipse/windowbuilder#752] [Q&A]
Hi

I'm new to this forum: I'm an italian teacher and in our laboratory we're experiencing WindowBuilder since few times.

I'd like to create a GUI that contains some simple geometric shapes: circle, rectangle, lines and so on.

In WindowBuilder I expected there should be a palette that contains such kind of objects (as it happens in VSC C#), but I haven't found it.
Is there a solution to use a WYSIWYG approach to manage these objects, with Drag&Drop insert and then "code behind" control?

Thank you in advance

Alberto

**Comment by ptziegler** [comment]
My apologies for the very late response. WindowBuilder is only able to build custom Swing/SWT composites but it doesn't allow you to "paint", for the lack of a better word.

I know that GEF has a Shapes example, which goes into that direction, but it's nowhere near practical usability and it also lacks the Java side. And I'm afraid that I don't know of any other tools that would provide such a complex functionality.
